gluten-core/src/main/scala/io/glutenproject/extension/columnar/RewriteTypedImperativeAggregate.scala

```diff
@@ -67,6 +100,28 @@ object RewriteTypedImperativeAggregate extends Rule[SparkPlan] with PullOutProje
         }
         copyBaseAggregateExec(agg)(newResultExpressions = newResultExpressions)
 
+      case agg: BaseAggregateExec
+          if agg.aggregateExpressions.exists(shouldRewriteForPercentileLikeExpr) =>
+        val exprMap = agg.aggregateExpressions
+          .filter(shouldRewriteForPercentileLikeExpr)
+          .map(ae => ae.aggregateFunction.inputAggBufferAttributes.head -> ae)
+          .toMap
+        val newResultExpressions = agg.resultExpressions.map {
+          case attr: AttributeReference =>
+            exprMap
+              .get(attr)
+              .map {
+                ae =>
+                  attr.copy(dataType = getPercentileLikeInterminateDataType(ae.aggregateFunction))(
+                    exprId = attr.exprId,
+                    qualifier = attr.qualifier
+                  )
+              }
+              .getOrElse(attr)
+          case other => other
+        }
+        copyBaseAggregateExec(agg)(newResultExpressions = newResultExpressions)
```

Comment:
IIUC this could make the offloaded partial aggregate operator incompatible with vanilla Spark's final aggregate operator? Say if final aggregation is fallen back then UB will be led.

I am considering whether we could using the approach like we had done for `approx_count_distinct`, to replace `approx_percentile` with something like `velox_approx_percentile` that has different intermediate type definition at the very early planning phase? Then we can distinguish between vanilla's and Velox's implementations for this function.

Reply:
ok, I'll rebase and refact it